sdk/src/builder.rs

```diff
@@ -593,13 +611,48 @@ impl Builder {
         Ok(self.definition.ingredients.last_mut().unwrap()) // ok since we just added it
     }
 
-    /// Adds an [`Ingredient`] to the manifest from an existing Ingredient.
-    pub fn add_ingredient<I>(&mut self, ingredient: I) -> &mut Self
+    /// Adds an [`Ingredient`] to the manifest from an existing Ingredient or JSON value.
+    ///
+    /// This method accepts any type that can be converted into an [`Ingredient`], including:
+    /// - An existing [`Ingredient`] instance
+    /// - A `serde_json::Value` containing ingredient data
+    /// - A JSON string (`&str` or `String`)
+    /// - Any other type that implements `TryInto<Ingredient>`
+    ///
+    /// # Arguments
+    /// * `ingredient` - The ingredient to add, either as an Ingredient or convertible type
+    /// # Returns
+    /// * A mutable reference to the [`Builder`].
+    /// # Errors
+    /// * Returns an [`Error`] if the ingredient cannot be converted or is invalid.
+    /// # Examples
+    /// ```rust
+    /// use c2pa::Builder;
+    /// use serde_json::json;
+    ///
+    /// let mut builder = Builder::new();
+    ///
+    /// // Add from JSON value
+    /// let ingredient_json = json!({
+    ///     "title": "My Ingredient",
+    ///     "relationship": "parentOf"
+    /// });
+    /// builder.add_ingredient(ingredient_json)?;
+    ///
+    /// // Add from JSON string
+    /// builder.add_ingredient(r#"{"title": "Another", "relationship": "componentOf"}"#)?;
+    /// # Ok::<(), c2pa::Error>(())
+    /// ```
+    pub fn add_ingredient<I>(&mut self, ingredient: I) -> Result<&mut Self>
     where
-        I: Into<Ingredient>,
+        I: TryInto<Ingredient>,
+        I::Error: std::fmt::Display,
```

Comment:
I'm leaning a bit more towards having `Builder.add_ingredient(Ingredient::from_json(json))` or something of the sort. I think this makes the most sense for Rust, but in our C bindings it would be fine if we had something like `c2pa_builder_add_ingredient(json)`